query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -196,6 +205,13 @@ public QueryCost canHandle(QueryRequest request, ExecutionContext executionConte
       cost = tableDefinition.getTimeGranularityMillis() / (Long.MAX_VALUE * 2D);
     }
 
+    long requestEndTime = getRequestEndTime(request.getFilter());
+    Duration requestDuration = Duration.ofMillis(requestEndTime - requestStartTime);
+
+    // check: requestDuration < minRequestDuration
+    if (!(requestDuration.compareTo(minRequestDuration) < 0)) {
```

Comment:
```suggestion
    if (requestDuration.compareTo(minRequestDuration) >= 0) {
```

Reply:
As we are selecting a handler based on minimum cost, here, would like to reduce the cost of the hander that can server has met.

In this `requestDuration.compareTo(minRequestDuration) >= 0`, I don't want to change anything.

Reply:
Oh! okey. I will change the boolean condition.

Reply:
Done, misunderstood with the ordering of handler.